### ISSUE TITLE
Set `fail` to `true` in pool session config leading to 'No resources available' error. 

### ISSUE THREAD

**Issue by hengfengli** [googleapis/nodejs-spanner#751]
#### Environment details

  - OS:macOS 10.14.6
  - Node.js version:v13.3.0
  - npm version:6.13.2
  - `@google-cloud/spanner` version:4.4.0

#### Steps to reproduce

index.ts: 
```typescript
/*
Table schema: 

CREATE TABLE Users (
    Name STRING(MAX),
    MySkills ARRAY<STRING(MAX)>,
    NeededSkills ARRAY<STRING(MAX)>,
    OfficeLocations ARRAY<STRING(MAX)>,
    PreferredDays STRING(MAX),
) PRIMARY KEY (Name)
*/
const { Spanner } = require('@google-cloud/spanner');

function readWriteTransaction(index, database) {
  return new Promise((resolve, reject) => {
    database.runTransaction(async (err, transaction) => {
      console.log("running i: ", index);

      if (err) {
        console.error(err);
        return;
      }
      const queryOne = {
        columns: ['Name', 'NeededSkills'],
        keySet: {
          all: true,
        },
      };
      const queryTwo = {
        columns: ['Name', 'MySkills'],
        keySet: {
          all: true,
        },
      };
  
      Promise.all([
        transaction.read('Users', queryOne).then(results => {
          const rows = results[0].map(row => row.toJSON());
          console.log(`NeededSkills: ${rows[0].NeededSkills}`);
        }),
  
        transaction.read('Users', queryTwo).then(results => {
          const rows = results[0].map(row => row.toJSON());
          console.log(`MySkills: ${rows[0].MySkills}`);
        }),
      ])
        .then(() => {
          // Updates the database
          // Note: Cloud Spanner interprets Node.js numbers as FLOAT64s, so they
          // must be converted (back) to strings before being inserted as INT64s.
          transaction.update('Users', [
            {
              Name: 'mdbuser/hengfeng',
              MySkills: ["test", "program"],
            },
          ]);
        })
        .then(() => {
          // Commits the transaction and send the changes to the database
          return transaction.commit();
        })
        .then(() => {
          console.log(
            `Successfully executed read-write transaction.`
          );
        })
        .catch(err => {
          console.error('ERROR:', err);
          reject(err);
        })
        .then(() => {
          transaction.end();  // cause an issue if not calling it, when an error happens. 
          resolve();
        });
    });
  });
};

const projectId = "e2e-debugging";
const instanceId = "hengfeng-test-instance";
const databaseId = "test-db";

async function runTest() {
  // Creates a client
  const spanner = new Spanner({
    projectId: projectId,
  });

  // Gets a reference to a Cloud Spanner instance and database
  const instance = spanner.instance(instanceId);
  const database = instance.database(databaseId, {
    max: 10,
    concurrency: 5,
    writes: 0.1,
    fail: true,
  });

  const promises = [];

  for (var _i = 0; _i < 10; _i++) {
    promises.push(readWriteTransaction(_i, database));
  }
  
  await Promise.all(promises);

  // Closes the database when finished
  console.log("closing the database connection.");
  database.close();
}

runTest();
```

#### Expected behaviour

No errors. When there is no available sessions, throw an error: 'No resources available.'

#### Actual behaviour

It directly throws errors from the beginning and the program is not working at all: 

```bash
...
running i:  9
Error: No resources available.
    at SessionPool._getSession (/Users/hengfeng/src/demo/cloud-spanner-js-demo/node_modules/@google-cloud/spanner/build/src/session-pool.js:486:19)
    at getSession (/Users/hengfeng/src/demo/cloud-spanner-js-demo/node_modules/@google-cloud/spanner/build/src/session-pool.js:272:40)
    at run (/Users/hengfeng/src/demo/cloud-spanner-js-demo/node_modules/p-queue/dist/index.js:249:104)
    at PQueue._tryToStartAnother (/Users/hengfeng/src/demo/cloud-spanner-js-demo/node_modules/p-queue/dist/index.js:197:38)
    at PQueue._next (/Users/hengfeng/src/demo/cloud-spanner-js-demo/node_modules/p-queue/dist/index.js:146:14)
    at run (/Users/hengfeng/src/demo/cloud-spanner-js-demo/node_modules/p-queue/dist/index.js:260:22)
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
```

**Comment by laljikanjareeya** [comment]
@hengfengli I think it's fair behavior as you can see in docs below for `SessionPoolOptions`:

https://github.com/googleapis/nodejs-spanner/blob/b891a811674faed13f3328d1304e0413256978f9/src/session-pool.ts#L116-L117

**Comment by hengfengli** [comment]
@laljikanjareeya Yes. I understand that. But it fails from beginning i = 0. 

If I set `fail` to true, the entire program fails and is not working. So what is the purpose of this `fail` option? 

Please try to run the code to see the effect.

**Comment by hengfengli** [comment]
I guess it is here: 

https://github.com/googleapis/nodejs-spanner/blob/b891a811674faed13f3328d1304e0413256978f9/src/session-pool.ts#L719-L726

At the beginning, `this.available` is 0 and if `fail` is set with `true`, it will just give an error "No resources available.".

**Comment by olavloite** [comment]
Fixed in #759
